artifacttemplates/http%3A%2F%2Fopentosca.org%2Fartifacttemplates/DockerEngine_DockerEngine-Interface-w1/source/src/main/java/org/opentosca/artifacttemplates/DockerClientHandler.java

```diff
@@ -0,0 +1,76 @@
+package org.opentosca.artifacttemplates;
+
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.model.Container;
+import com.github.dockerjava.api.model.ContainerPort;
+import com.github.dockerjava.api.model.Image;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientBuilder;
+
+public class DockerClientHandler {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DockerClientHandler.class);
+
+    protected static DefaultDockerClientConfig getConfig(String DockerEngineURL, String DockerEngineCertificate) {
+        DefaultDockerClientConfig.Builder config = DefaultDockerClientConfig
+                .createDefaultConfigBuilder()
+                .withDockerHost(DockerEngineURL)
+                .withApiVersion("1.21");
+
+        if (DockerEngineCertificate == null) {
+            config.withDockerTlsVerify(false);
+        } else {
+            config.withDockerCertPath(DockerEngineCertificate);
+        }
+        return config.build();
+    }
+
+    /**
+     * Check if the image is already available at the Docker host
+     *
+     * @param image the image to check availability for
+     * @return The image ID if the image is available, null otherwise
+     */
+    protected static String isImageAvailable(final String image, String dockerEngineURL, String dockerEngineCertificate) {
+        LOG.info("Searching available Images...");
+        DockerClient client = DockerClientBuilder.getInstance(getConfig(dockerEngineURL, dockerEngineCertificate)).build();
+        for (final Image availImage : client.listImagesCmd().exec()) {
+            // if there are 'none' images at the Docker Engine, e.g., from a local build, this results in null as availImage
```

Comment:
If there are *no* images

Reply:
The tricky thing is, that it only fails if you have images that show none as repository and tag if you execute `docker images`. So no images would be no problem.